### PR TITLE
Fix excessive path escaping in runtest py

### PR DESCRIPTION
**Pull request by jediry**
Running "python runtest.py SCons\Tools\SomeTest.py" fails on Windows with "PermissionError: [WinError 5] Access is denied". This is due to the call to escape() applied to args.python, resulting in a path like "C:\\\\Program Files\\\\Python310\\\\python.exe".

Removing the call to "escape()" resolves this failure. I have tested this fix on Windows 11 and Ubuntu.

## Contributor Checklist:

* [] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [] I have updated `CHANGES.txt` (and read the `README.rst`)
* [] I have updated the appropriate documentation

### CODE REVIEW

**Comment by mwichmann** [comment]
Thanks for dropping in!   There are indeed some issues with the handling of the strings used to specify tests - both on the command line and in test/exclude lists.  This hadn't been something I'd run into, though... can you think of a test that will trigger if the problem accidentally comes back? There's a `test/runtest` section it might fit into.

**Comment by bdbaddog** [comment]
You deleted the developer checklist for the PR.
I've restored it.
Please fill out the items you've completed and complete the items you haven't if they apply.

**Comment by bdbaddog** [comment]
Running 
`python .\runtest.py .\SCons\Tool\msvsTests.py`

Works fine for me without these changes.
@jediry - Can you provide an example which fails without this PR's changes?

File: testing/framework/TestRuntest.py [hunk 1]
```diff
@@ -51,11 +51,7 @@
     ]
 )
 
-if re.search(r'\s', python):
-    pythonstring = _python_
-else:
-    pythonstring = python
-pythonstring = pythonstring.replace('\\', '\\\\')
+pythonstring = python
```
Comment (mwichmann) [inline testing/framework/TestRuntest.py:54]: > __[Flake8]__ 'python' may be undefined, or defined from star imports: TestCommon ([view](https://sider.review/gh/repos/104670160/pulls/4218#issue-7724566))
>
> <details>
>
> | Rule |
> | ---- |
> | `F405` |
>
> You can [close this issue](https://sider.review/gh/repos/104670160/pulls/4218#issue-7724566) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

It is indeed from a star import: 
```python
from TestCommon import *
from TestCommon import __all__
```
so this one is okay.

**Pull request merged**